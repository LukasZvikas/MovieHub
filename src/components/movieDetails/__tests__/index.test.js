import { shallowMount, mount } from "@vue/test-utils";

import MovieDetailsWrapper from "../../movieDetails";

jest.mock("../../../utilities/fetch");

describe("SearchBar integration", () => {
  it("calls fetch", done => {
    const wrapper = mount(MovieDetailsWrapper);

    wrapper.vm.$nextTick(() => {
      wrapper.setData({
        movie_details: {
          id: 332562,
          poster_path: "wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg",
          title: "A Star Is Born",
          release_date: "2018-08-24"
        }
      });
      expect(wrapper.find('[data-test="movie-overview-title"]').text()).toBe(
        "A Star Is Born"
      );
      expect(
        wrapper.find('[data-test="movie-overview-release-date"]').text()
      ).toBe("(2018)");

      expect(wrapper.vm.$el).toMatchSnapshot();
      done();
    });
  });
});
