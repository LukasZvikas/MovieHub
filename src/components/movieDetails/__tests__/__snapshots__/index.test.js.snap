// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar integration calls fetch 1`] = `
<div
  class="movie-overview h-100"
>
  <div
    style="background-position: center; background-size: cover; display: flex; justify-content: center;"
  >
    <div
      class="movie-overview-description-width row"
    >
      <div
        class="col-sm-12 col-md-4 movie-overview-poster-wrap"
      >
        <img
          class="movie-overview-poster p-1 h-100 w-100"
          data-test="movie-overview-poster"
          src="http://image.tmdb.org/t/p/original/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
        />
      </div>
       
      <div
        class="col-sm-12 col-md-8 movie-overview-description p-2 d-flex justify-content-center"
      >
        <div
          class="movie-overview-title font-weight-bolder mb-2"
        >
          <h1
            data-test="movie-overview-title"
          >
            A Star Is Born
          </h1>
           
          <span
            class="font-weight-regular movie-overview-release-date"
            data-test="movie-overview-release-date"
          >
            (2018)
          </span>
        </div>
         
        <div
          class="row d-flex justify-content-start align-items-center"
        >
          <div
            class="col-sm-auto col-md-auto d-flex align-items-center mr-1 mb-4 mb-sm-0"
          >
            <h6
              class="font-weight-bold mr-1"
            >
              <div>
                User
              </div>
               
              <div>
                Score
              </div>
            </h6>
             
            <div>
              <svg
                class="circle-chart"
                height="70"
                viewBox="0 0 33.83098862 33.83098862"
                width="70"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="circle-chart__background"
                  cx="16.91549431"
                  cy="16.91549431"
                  fill="none"
                  r="15.91549431"
                  stroke="whitesmoke"
                  stroke-width="2"
                />
                 
                <circle
                  class="circle-chart__circle"
                  cx="16.91549431"
                  cy="16.91549431"
                  fill="none"
                  r="15.91549431"
                  stroke="#f70963"
                  stroke-dasharray="... 100"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                 
                <g
                  class="circle-chart__info"
                >
                  <text
                    alignment-baseline="central"
                    class="circle-chart__percent font-weight-bold"
                    fill="#fff"
                    font-size="8"
                    text-anchor="middle"
                    x="16.91549431"
                    y="16"
                  >
                    ...%
                  </text>
                </g>
              </svg>
            </div>
          </div>
           
          <div
            class="movie-overview-add-favorites d-flex justify-content-center align-items-center mr-1"
          >
            <svg
              class="svgclass"
              fill="#fff"
              height="20"
              version="1.1"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 0v32l10-10 10 10v-32z"
              />
            </svg>
          </div>
           
          <div
            class="movie-overview-add-favorites d-flex justify-content-center align-items-center"
          >
            <svg
              class="svgclass"
              fill="#fff"
              height="20"
              version="1.1"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.6 2c-3.363 0-6.258 2.736-7.599 5.594-1.342-2.858-4.237-5.594-7.601-5.594-4.637 0-8.4 3.764-8.4 8.401 0 9.433 9.516 11.906 16.001 21.232 6.13-9.268 15.999-12.1 15.999-21.232 0-4.637-3.763-8.401-8.4-8.401z"
              />
            </svg>
          </div>
        </div>
         
        <div
          class="mb-2"
        >
          <h3
            class="font-weight-bold mt-2"
          >
            Overview
          </h3>
           
          <div>
            
          </div>
        </div>
         
        <div
          class="row"
        />
      </div>
    </div>
  </div>
   
  <div
    style="margin: 2rem 0px;"
  >
    <div
      class="movie-overview-heading mb-1"
    >
      Top Billed Cast
    </div>
     
    <div
      class="row"
    />
  </div>
</div>
`;
